File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXInt8Value.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap the Int8 type and allows it to be null
 *
 */

public class TDBXInt8Value extends DBXValue {

	protected boolean ValueNull = false;
	private int DBXIntValue;
	
	public TDBXInt8Value() {
		super();
		setDBXType(DBXDataTypes.Int8Type);
	}
	
	public void setNull() {
		ValueNull = true;
		DBXIntValue = 0;
	}

	public boolean isNull() {
		return ValueNull;
	}
	
	public void SetAsInt8(int Value) throws DBXException {
		DBXIntValue = Value;
		ValueNull = false;
	}

	public int GetAsInt8() throws DBXException {
		return DBXIntValue;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TDBXInt8Value.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXInt8Value.h"


@implementation TDBXInt8Value
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:Int8Type];
	}
	return self;
}
-(void) dealloc{
	[super dealloc];
}
-(void) SetNull {
	ValueNull = YES;
	DBXInternalValue = 0;
}
-(bool) isNull {
	return ValueNull;
}
-(void) SetAsInt8:(int)value{
	ValueNull = NO;
	DBXInternalValue = value;
	
}
-(int) GetAsInt8{
	return DBXInternalValue;
}

@end
